lldb/source/Host/macosx/objcxx/HostInfoMacOSX.mm

```diff
@@ -144,16 +150,22 @@ static void ParseOSVersion(llvm::VersionTuple &version, NSString *Key) {
 #endif
   } else {
     // Find the bin path relative to the lib path where the cmake-based
-    // OS X .dylib lives.  This is not going to work if the bin and lib
-    // dir are not both in the same dir.
+    // OS X .dylib lives. We try looking first at a possible sibling `bin`
+    // directory, and then at the `lib` directory itself.
```

Comment:
If you're doing this to support non-CMake build systems (e.g. Bazel), could you write some details explaining why we might want to look in `lib/`?  I was thinking something to the effect of "some build systems may put executables in the same directory as the final liblldb so we look there too".